### PR TITLE
Future-proof detect-minions for gke

### PR DESCRIPTION
**Pull request by j3ffml**
This will fix failing gke-ci jenkins build
cc @saad-ali @ixdy @zmerlynn

### CODE REVIEW

File: cluster/gke/util.sh [hunk 1]
```diff
@@ -241,9 +241,15 @@ function detect-minion-names {
   echo "... in gke:detect-minion-names()" >&2
   detect-project
   detect-node-instance-group
-  MINION_NAMES=($(gcloud compute instance-groups managed list-instances \
-    "${NODE_INSTANCE_GROUP}" --zone "${ZONE}" --project "${PROJECT}" \
-    | cut -d'/' -f11))
+  if [[ ! -z "${DOGFOOD_GLOUD:-}" ]]; then
```
Comment (ixdy) [inline cluster/gke/util.sh]: Is using this env variable the right choice? Will this eventually affect the stable release of gcloud?

Comment (jlowdermilk) [inline cluster/gke/util.sh]: Actually not needed. This command will work fine with the current version.

**Comment by ixdy** [comment]
I assume you've tested this? LGTM

File: cluster/gke/util.sh [hunk 2]
```diff
@@ -243,7 +243,8 @@ function detect-minion-names {
   detect-node-instance-group
   MINION_NAMES=($(gcloud compute instance-groups managed list-instances \
     "${NODE_INSTANCE_GROUP}" --zone "${ZONE}" --project "${PROJECT}" \
-    | cut -d'/' -f11))
+    --format=yaml | grep instances: | cut -d ' ' -f 2))
```
Comment (ixdy) [inline cluster/gke/util.sh]: I think this needs to be `instance:` (not plural)

**Comment by ixdy** [comment]
cancel LG

**Comment by ixdy** [comment]
I tested the command manually on jenkins-master, LG.

**Comment by ixdy** [comment]
@saad-ali the presubmit tests aren't going to test anything in this, recommend merging ASAP to stop the bleeding

**Comment by roberthbailey** [comment]
Do we need to fix this in `cluster/gce/util.sh` as well? https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L250-L252

**Comment by jlowdermilk** [comment]
Yeah, we need to fix it in gce as well. Pretty sure that's what's breaking it now.

**Pull request merged**